### DISCUSSION TITLE
LightMode background color still visible with "darkMode: class" config

### DISCUSSION THREAD

**Discussion by pingustar** [tailwindlabs/tailwindcss#10096] [Help] [answered]
**What version of Tailwind CSS are you using?**

v3.2.4

**What build tool (or framework if it abstracts the build tool) are you using?**

vite 4.0.1, react 18.2.0

**What version of Node.js are you using?**

v16.16.0

**What browser are you using?**

Chrome, Brave

**What operating system are you using?**

macOS

**Reproduction URL**

https://github.com/pingustar/tailwind-bug-reproduction

**Describe your issue**

When using `darkMode: "class"` in my tailwind config, the light mode background color is still visible on dark mode.

Even when setting `min-h-screen` to `html, body, #root`, the scroll bounce unveils the light mode background color.

When using `darkMode: "media"` in my tailwind config, everything works as expected.

https://user-images.githubusercontent.com/7262066/207910629-b0dda33b-0cd1-4c1b-9f99-90a0fbb758f1.mov

**Comment by RobinMalfait** [comment, marked as answer]
Hey! Thank you for your bug report!
Much appreciated! 🙏

The issue here is that the `dark:x` variant is implemented as `.dark &`. This means that if you do something like `html { @apply dark:x }` that the selector will be `.dark html {}` this can never work because there is nothing above the `html`.

One way _we_ could fix it is by using `.dark &, &.dark {}` but that would create many unnecessary selectors because this is only useful to the `html`.

Instead, I would recommend you to put the darkmode class on the `html` tag, and update your css to:
```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
    body, #root {
        @apply bg-blue-500 dark:bg-red-500;
    }
    html.dark { @apply bg-red-500; }
}
```

Hope that helps!
